packages/gdl-frontend/components/EditBookLibrary/BookSelectionLink.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+
+import * as React from 'react';
+import { CardContent, Typography } from '@material-ui/core';
+import { CheckCircle } from '@material-ui/icons';
+import styled from '@emotion/styled';
+
+import CoverImage from '../CoverImage';
+import media from '../../style/media';
+import { coverWidths } from '../BookListSection/coverWidths';
+
+export type Book = $ReadOnly<{
+  id: string,
+  bookId: number,
+  title: string,
+  language: {
+    code: string
+  },
+  coverImage: ?{ url: string }
+}>;
+
+type Props = {
+  book: Book,
+  selectedBooks: Array<string>,
+  active: number,
+  changeActive: () => void
+};
+
+type State = {
+  active: boolean
+};
+
+export default class BookLink extends React.Component<Props, State> {
+  state = {
+    active: false
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.active !== this.props.active) {
+      if (this.props.active === 0) {
+        this.setState({ active: false });
+      } else if (this.props.active === 1) {
+        this.setState({ active: true });
+      }
+    }
+  }
+  handleClick(id: string, selectedBooks: Array<string>) {
+    if (!selectedBooks.some(item => id === item)) {
+      selectedBooks.push(id);
+      this.setState({ active: true });
+    } else {
+      selectedBooks.splice(selectedBooks.indexOf(id), 1);
+      this.setState({ active: false });
+    }
+    this.props.changeActive();
+  }
+
+  render() {
+    const { book, selectedBooks } = this.props;
+    if (this.state.active && !selectedBooks.some(item => book.id === item)) {
```

Comment:
kanskje trekke ut `selectedBooks.some(item => book.id === item)` i en `const` også bruke den i de to uttrykkene så blir det litt lettere å lese koden :)